src/core/eme/utils/persistent_sessions_store.ts

```diff
@@ -142,6 +145,37 @@ export default class PersistedSessionsStore {
     this._save();
   }
 
+  /**
+   * Retrieve index of an entry.
+   * Returns `-1` if not found.
+   * @param {Uint8Array}  initData
+   * @param {string|undefined} initDataType
+   * @returns {number}
+   */
+  private getIndex(
+    initData : Uint8Array,
+    initDataType : string|undefined
+  ) : number {
+    const hash = hashBuffer(initData);
+    for (let i = 0; i < this._entries.length; i++) {
+      const entry = this._entries[i];
+      if (entry.initDataType === initDataType) {
+        if (entry.version === 1) {
+          if (entry.initDataHash === hash) {
+            if (areArraysOfNumbersEqual(entry.initData, initData)) {
```

Comment:
Why wouldn't make a unique `if` ?

Reply:
Yep, I don't know why I've written it that way, thanks

Reply:
done